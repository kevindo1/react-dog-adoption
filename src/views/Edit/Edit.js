import React from 'react';
import { useState, useEffect } from 'react';
import { fetchDogsById, updateDog } from '../../services/dogs';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
import DogForm from '../../components/DogForm';

export default function Admin() {
  const [dog, setDog] = useState({});
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogsById(params.id);
      setDog(data);
    };
    fetchData();
  }, [params.id]);

  // const updateDogState = (key, value) => {
  //   dog[key] = value;
  //   setDog({ ...dog });
  // };

  const handleSubmit = async (e) => {
    e.preventDefault;
    await updateDog(dog);
  };

  return (
    <>
      <div>
        {/* <DogForm dog={dog} handleSubmit={handleSubmit} updateDogState={updateDogState} /> */}
      </div>
    </>
  );
}
