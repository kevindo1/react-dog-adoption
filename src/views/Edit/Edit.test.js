import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom/cjs/react-router-dom.min';
import Edit from './Edit';

test('renders Edit individual dog page', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/12/edit']}>
      <Route path="/dogs/:id/edit" component={Edit} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Thor');

  expect(container).toMatchSnapshot();
});
