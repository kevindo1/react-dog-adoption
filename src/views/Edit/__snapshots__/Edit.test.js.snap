// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Edit individual dog page 1`] = `
<div>
  <div>
    <div>
      <form>
        <label>
          Name:
          <input
            name="name"
            type="text"
            value="Thor"
          />
        </label>
        <label>
          Age:
          <input
            name="age"
            type="text"
            value="4"
          />
        </label>
        <label>
          Image url:
          <input
            name="image"
            type="text"
            value="https://placedog.net/500?id=3"
          />
        </label>
        <label>
          Breed:
          <input
            name="breed"
            type="text"
            value="Asguardian"
          />
        </label>
        <label>
          Bio:
          <input
            name="bio"
            type="text"
            value="God of Thunder"
          />
        </label>
        <div>
          <button>
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
